Collective selection infrastructure update (get rid of explicit casting, namespacing on CH4 level correction)

AlexeyMalkhanov opened the pull request:
The PR contains the following changes:
1) get rid of explicit container types casting in selection functions by introducing ```const MPIDI_< |OFI|POSIX>_coll_algo_container_t *``` instead of ```MPIDI_< |OFI|POSIX>_coll_algo_container_t *``` 
2) Namespacing prefixes correction on CH4 layer

wesbland commented:
There's a bunch of formatting problems.

AlexeyMalkhanov commented:
Formatting problems have been corrected

AlexeyMalkhanov commented:
@raffenet @wesbland it seems that the pre-commit hook doesn't take into account the state of the source in the repository. This is the reason why whitespace checker reports issues. If I apply these changes we'll get lines in the difference which are not a part of the actual PR.

wesbland commented:
Have you rebadged onto the latest master since we merged the formatting fix again? That should improve the baseline.

On Mar 14, 2018, at 7:16 AM, AlexeyMalkhanov <notifications@github.com<mailto:notifications@github.com>> wrote:


@raffenet<https://github.com/raffenet> @wesbland<https://github.com/wesbland> it seems that the pre-commit hook doesn't take into account the state of the source in the repository. This is the reason why whitespace checker reports issues. If I apply these changes we'll get lines in the difference which are not a part of the actual PR.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/pmodels/mpich/pull/3016#issuecomment-373000669>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABfSd1CWzM6JXTZpgbldJJEHiyyoc2Ucks5teQoegaJpZM4SZ2FG>.

AlexeyMalkhanov commented:
@wesbland  I've made my changes on top of c8d7179788e2d94883a5a4ffc6defadcfaaf05b2 commit. The current master contains two more commits but they do not relate formatting.

wesbland commented:
So the thing that the formatting checker doesn't like is that your changes took out some casts but left the space between the `&` and the return value. Also adding `const` made it want to un-indent the function signatures for some reason.

Anyway, I'll just fix it. Make sure you have the pre-commit hook set up so this problem can be avoided in the future.

raffenet commented on src/mpid/ch4/include/coll_algo_params.h.in:
```diff
@@ -12,135 +12,135 @@
 typedef union MPIDI_Barrier_algo_params {
     /* *INDENT-OFF* */
-    MPIDI_CH4_BARRIER_PARAMS_DECL;
+    MPIDI_CH4I_BARRIER_PARAMS_DECL;
```
At some point we were in agreement that `MPIDI_CH4I` is redundant given that only one device is built at a time. Therefore all of these instances can be renamed `MPIDI_CH4I` -> `MPIDI`. Can you make that change instead?

AlexeyMalkhanov commented:
@raffenet I've introduced the requested changes. In order to distinguish structures from former CH4 layer and structures of parameters which comprise all layers, for the latter  I've added prefix _COMMON_.

raffenet commented on src/mpid/ch4/src/ch4_coll_select.h:
```diff
@@ -20,71 +20,68 @@
+MPIDI_coll_algo_container_t *MPIDI_CH4_Barrier_select(MPIR_Comm * comm, MPIR_Errflag_t * errflag)
 {
     if (comm->comm_kind == MPIR_COMM_KIND__INTERCOMM) {
-        return (MPIDI_coll_algo_container_t *) & MPIDI_CH4I_Barrier_inter_composition_alpha_cnt;
```
I agree with the removal of these casts, but that's not what is described by this commit message. IMO, they are a different logical change and should be a separate commit.

raffenet commented on src/mpid/ch4/include/coll_algo_params.h.in:
```diff
@@ -8,186 +8,186 @@
+#define MPIDI_COLL_AUTO_SELECT (-1)
 
-typedef union MPIDI_Barrier_algo_params {
+typedef union MPIDI_COMMON_Barrier_algo_params {
```
I don't think adding COMMON here is the right thing to do. Maybe our "generic" prefix is more suitable? I.e. `MPIDIG_`. I'm not sure.

raffenet replied:
Maybe we can just drop this change if it won't cause issues or collisions.

AlexeyMalkhanov replied:
@raffenet after the removal CH4 prefix the names for the union on the CH4 level and union which comprises all layers became identical. That is why we have decided to add prefix COMMON. So, unfortunately, we can't just drop this change. I will add prefix ```MPIDIG_``` instead.

raffenet commented:
test:jenkins/ch4/most

raffenet commented:
test:jenkins/ch4/most
